### PR TITLE
apps:nettest modify multicast test in nettest

### PR DESCRIPTION
**Pull request by pillip8282**
1) Add an interface name in command line to get an allocated ip address from system
2) Modify an usage because old version is difficult to understand
3) Bug fix in netlib_getipv4addr: NET_LWIP should not be used in netlib

### CODE REVIEW

File: apps/examples/nettest/nettest.c [hunk 1]
```diff
@@ -650,67 +731,83 @@ int main(int argc, FAR char *argv[])
...
-	if (argc < 4) {
-		printf("\n\nUsage1: nettest target_addr target_port [0(tcpserv -infinite pkt receive), 1(tcpserv), 2(udpserv), 3(ipmulticastreceiver)]\n\n");
-		printf("\n\nSample Command1 (for ipmulticastreceiver): nettest 225.1.1.1 5555 3\n\n");
-		printf("\n\n");
-		printf("\n\nUsage2: nettest target_addr target_port [4(tcpclient), 5(udpclient), 6(ipmulticastsender)] pps(optional, number of packets transmitted per second - default value 1)\n\n");
-		printf("\n\nSample Command2 (for udpclient): nettest 192.168.2.5 5555 5 4\n\n");
-		printf("\n\n");
-		return 0;
+
+	if (argc < 5) {
```
Comment (davidfather) [inline apps/examples/nettest/nettest.c]: Why not 6?

File: apps/examples/nettest/nettest.c [hunk 2]
```diff
@@ -650,67 +731,83 @@ int main(int argc, FAR char *argv[])
...
+		proto = NT_TCP;
+	} else if (!strncmp(argv[2], NETTEST_PROTO_UDP, strlen(NETTEST_PROTO_UDP) + 1)) {
+		proto = NT_UDP;
+	} else if (!strncmp(argv[2], NETTEST_PROTO_MULTICAST, strlen(NETTEST_PROTO_MULTICAST) + 1)) {
+		proto = NT_MULTICAST;
+	} else {
+		goto err_with_input;
+	}
+
+	if (argc < 6) {
```
Comment (davidfather) [inline apps/examples/nettest/nettest.c]: Is it duplicated with the line 741?

File: apps/examples/nettest/nettest.c [hunk 3]
```diff
@@ -650,67 +731,83 @@ int main(int argc, FAR char *argv[])
 int nettest_main(int argc, char *argv[])
 #endif
 {
-	nlldbg("Running nettest_main \n");
-	int proto = 0;
-	int num_packets_to_process = NUM_PACKETS;
-	uint32_t sleep_time = 0;
+	int mode = 0;
+	nettest_proto_e proto = 0;
```
Comment (davidfather) [inline apps/examples/nettest/nettest.c]: Does this line mean that the default value of proto is NT_TCP?

File: apps/examples/nettest/nettest.c [hunk 4]
```diff
@@ -199,64 +262,75 @@ void ipmcast_sender_thread(int num_packets, uint32_t sleep_time)
...
-		sleep(1);
 	}
 
-	printf("\n[MCASTCLIENT] Terminate multicast sender after sending sufficient messages (%d)\n", num_packets);
+	printf("[MCASTCLIENT] Terminate multicast sender after sending sufficient messages (%d)\n", num_packets);
 	close(sd);
 	return;
 
 errout_with_socket:
 	close(sd);
```
Comment (davidfather) [inline apps/examples/nettest/nettest.c]: Please, remove these two lines and move 'errout_with_socket' between line 299 and line 300.

File: apps/examples/nettest/nettest.c [hunk 5]
```diff
@@ -199,64 +262,75 @@ void ipmcast_sender_thread(int num_packets, uint32_t sleep_time)
...
 	struct sockaddr_in localSock;
 	struct ip_mreq group;
-	printf("\n[MCASTSERV] start multicast receiver\n");
-	/* ------------------------------------------------------------ */
-	/*                                                              */
-	/* Receive Multicast Datagram code example.                     */
-	/*                                                              */
-	/* ------------------------------------------------------------ */
+	printf("[MCASTSERV] start multicast receiver\n");
+
```
Comment (davidfather) [inline apps/examples/nettest/nettest.c]: Please, remove this empty line.

File: apps/examples/nettest/nettest.c [hunk 6]
```diff
@@ -307,24 +392,26 @@ void ipmcast_receiver_thread(int num_packets)
...
+		}
+		if (count > num_packets) {
+			printf("[MCASTSERV] Terminating MCLIENT as received sufficient packets for testing\n");
+			break;
 		}
 	}
 	close(sd);
 	return;
 
 errout_with_socket:
```
Comment (davidfather) [inline apps/examples/nettest/nettest.c]: Please, remove two lines (413~414) and move 'errout_with_socket' between line408 and line 409.

File: apps/examples/nettest/nettest.c [hunk 7]
```diff
@@ -507,63 +592,57 @@ void tcp_server_thread(int num_packets, int infpkt)
...
+			printf("Exiting tcp_server as received sufficient packets for testing\n");
+			break;
 		}
 	}
 	if (connfd > 0) {
 		close(connfd);
-		printf("\n[TCPSERV] Closed connfd successfully \n");
+		printf("[TCPSERV] Closed connfd successfully \n");
 	}
 	close(listenfd);
```
Comment (davidfather) [inline apps/examples/nettest/nettest.c]: Please, remove redundant code from line 639 to line 645.

File: apps/examples/nettest/nettest.c [hunk 8]
```diff
@@ -590,57 +669,59 @@ void tcp_client_thread(int num_packets, uint32_t sleep_time)
...
+		printf("[TCPCLIENT] <--send\t%d bytes\n\n", ret);
+		if (sleep_time != 0) {
+			usleep(sleep_time);
+		}
 	}
 
-	printf("\n[TCPCLIENT] Terminating tcpclient after sending sufficient messages (%d)\n", num_packets);
+	printf("[TCPCLIENT] Terminating tcpclient after sending sufficient messages (%d)\n", num_packets);
 
 	close(sockfd);
```
Comment (davidfather) [inline apps/examples/nettest/nettest.c]: Please, remove redundant code from line 719 to line 724.

File: apps/examples/nettest/nettest.c (hunk 1 above)
Comment (pillip8282) [reply apps/examples/nettest/nettest.c]: I was confused, I will fix

File: apps/examples/nettest/nettest.c (hunk 2 above)
Comment (pillip8282) [reply apps/examples/nettest/nettest.c]: yes, it's duplicated

File: apps/examples/nettest/nettest.c (hunk 3 above)
Comment (pillip8282) [reply apps/examples/nettest/nettest.c]: I will add an addtional field.

File: apps/examples/nettest/nettest.c (hunk 4 above)
Comment (pillip8282) [reply apps/examples/nettest/nettest.c]: ok

File: apps/examples/nettest/nettest.c (hunk 5 above)
Comment (pillip8282) [reply apps/examples/nettest/nettest.c]: ok

**Comment by davidfather** [comment]
Still one redundant part is remained around line number 712~717.

**Pull request merged**